src/aggregation/AggregatorFactory.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IdentifiedDataSerializable, IdentifiedDataSerializableFactory} from '../serialization/Serializable';
+import {
+    Aggregator,
+    CountAggregator,
+    DoubleAverageAggregator,
+    DoubleSumAggregator,
+    FixedPointSumAggregator,
+    FloatingPointSumAggregator,
+    IntegerAverageAggregator,
+    IntegerSumAggregator,
+    LongAverageAggregator,
+    LongSumAggregator,
+    MaxAggregator,
+    MinAggregator,
+    NumberAverageAggregator
+} from './Aggregator';
+import {LoggingService} from '../logging/LoggingService';
+
+export class AggregatorFactory implements IdentifiedDataSerializableFactory {
+
+    private logger = LoggingService.getLoggingService();
+    static readonly FACTORY_ID = -41;
+
+    static readonly BIG_DECIMAL_AVG = 0; //not implemented in node.js
+    static readonly BIG_DECIMAL_SUM = 1; //not implemented in node.js
+    static readonly BIG_INT_AVG = 2; //not implemented in node.js
+    static readonly BIG_INT_SUM = 3; //not implemented in node.js
+    static readonly COUNT = 4;
+    static readonly DISTINCT = 5; //returns java serializable, not usable in node.js
```

Comment:
Was this because serialization can't handle Set?

Reply:
`DistinctValuesAggregator.aggregate()` on Java side returns `Set`. Therefore client receives a Java `Set` as response from the server.